enhancements/network/host-port-registry.md

```diff
@@ -23,35 +23,71 @@ This document serves as the authoritative list of port assignments.
+conflicts.
 
-The user-facing documentation, informed by the installer's
-terraform templates, is the authoritative source for usable port ranges.
+OCP reserves host ports 9000-9999 (both TCP and UDP) for use by system
+services that communicate between nodes. Ports 29000-29999 (both TCP
+and UDP) are reserved for use by system services that listen on
+(host-network) localhost only. (OCP does not reserve any SCTP ports.)
 
-These ports are used by certain services that, for whatever reason, must run
-in the host network (or use host-port forwarding).
+Additionally, some services that have standard port assignments run on
+ports outside this range, either for historical reasons, or to allow
+them to be more easily recognized by network debugging and monitoring
+tools.
 
-The Kubernetes scheduler is aware of host ports, and will fail to schedule pods
-on a node where there would be a port conflict.
+Other than the reserved ranges and the other ports listed below, host
+ports on worker nodes are available for use by customers. (But all
+host ports on masters are reserved for OCP use.)
 
-### Localhost
+The Kubernetes scheduler is aware of host ports, and will fail to
+schedule pods on a node where there would be a port conflict.
 
-Separately, it is a common pattern for a pod to have one or more processes only
-listening on localhost. Since all HostNetwork pods share a network namespace,
-coordination is also required in this case.
+## Configuration
 
-## Requirements
+The installer's terraform templates open the 9000-9999 range [and
+several
+others](https://github.com/openshift/openshift-docs/blob/master/modules/installation-network-user-infra.adoc)
+between nodes in IPI clusters. Customers are required to ensure these
+ranges are open in UPI clusters. The user-facing documentation,
+informed by the installer's terraform templates, is the authoritative
+source for reserved port ranges.
+
+## Requirements for OCP Components
 
 All processes that wish to listen on a host port MUST
 
-- Have an entry in the table below
+- have an entry in the table below
 - be in a [documented range](https://github.com/openshift/openshift-docs/blob/master/modules/installation-network-user-infra.adoc),
-  if they are intended to be reachable
+  if they are intended to be reachable from other nodes
 - declare that port in their `Pod.Spec`
 
-Localhost-only ports SHOULD be outside of this range, to leave room.
+Starting with OCP 4.8, all new worker-node host port allocations
+SHOULD be in the reserved 9000-9999 port range, and all new
+worker-node localhost-only port allocation SHOULD be in the
+29000-29999 range, to avoid conflicting with ports claimed by customer
+pods created in earlier releases. If there is a strong argument for
+using a port outside these ranges by default, the relevant operator
+MUST have a configuration option to allow a customer to move the
+service into the reserved port range (or to a port of the customer's
+choice) instead.
```

Comment:
It seems like we would only want the customer to move it to a port that is not part of a reserved range. Otherwise we'll have to deal with upgrades where a customer has moved a component to a reserved port that is unused in 4.N but being used by a new service added in 4.N+1.

Reply:
I meant that for any given non-reserved port we use in the future, we'd also have a _specific_ reserved port to go with it. eg, the service runs on 12345 by default, but there's a config option to flip it to 22345 instead, and both of those ports would be listed as reserved for that service in `host-port-registry.md`

Reply:
OK, that makes sense but isn't really clear from what's written. An example, as you gave in the comment, would help.